citrination_client/data/tests/test_data_client.py

```diff
@@ -59,6 +59,15 @@ def test_upload_pif():
     with open("tmp.json", "r") as fp:
         assert json.loads(fp.read())["uid"] == pif.uid
 
+def test_does_not_require_trailing_slash():
+    src_path = "{}data_holder".format(test_file_data_root)
+    result = client.upload(dataset_id, src_path)
+    assert result.successful()
+
+    files = client.list_files(dataset_id)
+    first_path = files[0]
+    assert "data_holder/data_holder" not in first_path
```

Comment:
The `files[0]` followed by a default-negative check seems easy to pass by accident.  What's to say this is the first file in the dataset?

Reply:
If any of them have that pattern, they will all have that pattern.

Reply:
(this is part of the path, not the file name)